server/grpc_service.go

```diff
@@ -1000,7 +1000,7 @@ func (s *GrpcServer) GetRegionByID(ctx context.Context, request *pdpb.GetRegionB
 		return &pdpb.GetRegionResponse{Header: s.header()}, nil
 	}
 	var buckets *metapb.Buckets
-	if request.GetNeedBuckets() {
+	if request.GetNeedBuckets() && rc.GetStoreConfig().EnableRegionBucket() {
```

Comment:
move it before `request.GetNeedBuckets()?`

Reply:
fix